### ISSUE TITLE
Add preprocessor switch to disable file system include / support

### ISSUE THREAD

**Issue by W4RH4WK** [fmtlib/fmt#4257]
This is a feature request for adding a preprocessor switch that allows one to disable the inclusion of the `std::filesystem` header in `std.h`.

The problem is that some video game console SDKs do not support `std::filesystem`, but ship the `<filesystem>` header. Therefore `FMT_HAS_INCLUDE` returns true, but the `<filesystem>` header provided by the SDK contains an `#error` directive breaking the build.

I'd appreciate the inclusion of a preprocessor switch (e.g. `FMT_DISABLE_STD_FILESYSTEM`) which can be set prior including `std.h` to work around this issue.

**Comment by vitaut** [comment]
> some video game console SDKs do not support `std::filesystem`, but ship the `<filesystem>` header

Could you provide some details about which SDKs do that?

**Comment by W4RH4WK** [comment]
> Could you provide some details about which SDKs do that?

The Nintendo Switch SDK.

**Comment by vitaut** [comment]
{fmt} already has `FMT_CPP_LIB_FILESYSTEM` that wraps the corresponding feature-test macro and is overridable: https://github.com/fmtlib/fmt/blob/47a66c5eccc0cce71ad81b4a681a2032d86ca951/include/fmt/std.h#L65-L71 Does defining it to zero address your use case or is anything else missing?

**Comment by W4RH4WK** [comment]
The problem originates from `#include <filesystem>` here:
https://github.com/fmtlib/fmt/blob/47a66c5eccc0cce71ad81b4a681a2032d86ca951/include/fmt/std.h#L29-L31

This block is above the `FMT_CPP_LIB_FILESYSTEM`, so defining `FMT_CPP_LIB_FILESYSTEM=0` does not solve the problem.

**Comment by W4RH4WK** [comment]
The SDK's `<filesystem>` header includes this line (as `std::filesystem` is not supported) which breaks the build. That's why I'd like to prevent fmt from even including the header.

```
# error "The <filesystem> library is not supported since libc++ has been configured without support for a filesystem."
```

**Comment by vitaut** [comment]
I think we can move the `FMT_CPP_LIB_FILESYSTEM` before the `<filesystem>` include (it only needs `<version>`) and make the include conditional on it. A PR would be welcome.

**Comment by vitaut** [comment]
Addressed by #4258.
